x/gamm/pool-models/stableswap/amm.go

```diff
@@ -320,11 +320,14 @@ func (p *Pool) calcSingleAssetJoinShares(tokenIn sdk.Coin, swapFee sdk.Dec) (sdk
 // We can mutate pa here
 // TODO: some day switch this to a COW wrapped pa, for better perf
 func (p *Pool) joinPoolSharesInternal(ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, newLiquidity sdk.Coins, err error) {
-	if len(tokensIn) == 1 {
+	if !tokensIn.DenomsSubsetOf(p.GetTotalPoolLiquidity(ctx)) {
+		return sdk.ZeroInt(), sdk.NewCoins(), errors.New("attempted joining pool with assets that do not exist in pool")
+	}
+	if len(tokensIn) == 1 && tokensIn[0].Amount.GT(sdk.OneInt()) {
```

Comment:
Why do we have these `coin.Amount.GT(sdk.OneInt())` checks?

Reply:
1 token single-asset joins lead to a division by zero since we truncate at int bounds during swaps. With these checks, 1 token single-asset joins are treated as 0 token joins, which I assumed was acceptable given our rounding assumptions

(This came up in this PR because I happened to catch it in a test case I added)